metadata-ingestion/src/datahub/ingestion/source/unity/ge_profiler.py

```diff
@@ -143,6 +144,17 @@ def get_unity_profile_request(
                 self.report.report_dropped(dataset_name)
             return None
 
+        if profile_table_level_only and table.size_in_bytes is not None:
+            # For requests with profile_table_level_only set, dataset profile is generated
+            # by looking at table.rows_count. For delta tables (a typical databricks table)
+            # count(*) is an efficient query to compute row count.
+            # Presence of size_in_bytes confirms this is DELTA table and that we have
+            # SELECT permission on this table.
```

Comment:
I wonder if we may transform this comment in actual code. Something like:

```
is_delta_table = table.size_in_bytes is not None
```

and then use the new flag accordingly to call delta-specific code

WDYT?

Reply:
In fact, we can add `is_delta_table` boolean in UnityCatalogSQLGenericTable based on custom property `data_source_format` in original table.
Or we can add `is_delta_table_with_select_enabled: table.size_in_bytes is not None` as suggested. Maybe I can do both.